library/std/src/sys/windows/api.rs

```diff
@@ -0,0 +1,51 @@
+use crate::cmp;
+use crate::ffi;
+use crate::os::windows::prelude::{AsRawHandle, BorrowedHandle};
+use crate::sys::c;
+use core::mem::MaybeUninit;
+use core::ptr::null_mut as null;
+
+#[derive(Copy, Clone)]
+pub(super) struct Error(pub u32);
+impl Error {
+    pub fn code(self) -> u32 {
+        self.0
+    }
+}
+
+/// Read up to `u32::MAX` bytes from a file.
+///
+/// If `overlapped` is `None` then it reads from the current file offset.
+/// Otherwise the `Offset` and `OffsetHigh` fields of `OVERLAPPED` indicate the
+/// file position. These will be ignored if the file does not support offsets.
+///
+/// # Safety
+///
+/// If the handle is asynchronous then `buffer` and `overlapped` must live
+/// at least until the read operation completes. Which may be well after this
+/// function returns.
+///
+/// If the handle is known to be synchronous then this function is safe.
+pub(super) unsafe fn ReadFile<'handle, 'overlapped>(
```

Comment:
This seems fine but it's also a bit weird to make this change at the same time. Am I missing why it's needed? (Probably error handling?)

Reply:
Yeah this is me being a bit extra, sorry. I can save it for another day if you'd prefer.